kv/batch.go

```diff
@@ -40,6 +40,8 @@ func newRSpan(ba roachpb.BatchRequest) rSpan {
 
 // intersect returns the intersection of the current span and the
 // descriptor's range.
+// TODO(kkaneda): Currently the method returns desc when desc and the span
+// do not overlap. Fix if this is not the expected behavior here.
```

Comment:
Sure, changed the code to return an error.

I also removed the check for `rs.endKey == nil`. It seems that won't happen since `rSpan` is created only from `keys.Range()`